src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -204,1024 +204,1046 @@ internal void OnBeforeWrite(TokenCacheNotificationArgs args)
+                            "Access token is expired.  IsExtendedLifeTimeEnabled=TRUE and ExtendedExpiresOn is not exceeded.  Returning the found cache entry. " +
+                            GetAccessTokenExpireLogMessageContent(msalAccessTokenCacheItem));
+
+                        msalAccessTokenCacheItem.IsExtendedLifeTimeToken = true;
+                        return msalAccessTokenCacheItem;
+                    }
+
                     requestParams.RequestContext.Logger.Info(
-                        "Access token is expired.  IsExtendedLifeTimeEnabled=TRUE and ExtendedExpiresOn is not exceeded.  Returning the found cache entry. " +
+                        "Access token has expired or about to expire. " +
                         GetAccessTokenExpireLogMessageContent(msalAccessTokenCacheItem));
-
-                    msalAccessTokenCacheItem.IsExtendedLifeTimeToken = true;
-                    return msalAccessTokenCacheItem;
                 }
 
-                requestParams.RequestContext.Logger.Info(
-                    "Access token has expired or about to expire. " +
-                    GetAccessTokenExpireLogMessageContent(msalAccessTokenCacheItem));
+                return null;
             }
-
-            return null;
         }
-    }
 
-    private string GetAccessTokenExpireLogMessageContent(MsalAccessTokenCacheItem msalAccessTokenCacheItem)
-    {
-        return string.Format(
-            CultureInfo.InvariantCulture,
-            "[Current time ({0}) - Expiration Time ({1}) - Extended Expiration Time ({2})]",
-            DateTime.UtcNow,
-            msalAccessTokenCacheItem.ExpiresOn,
-            msalAccessTokenCacheItem.ExtendedExpiresOn);
-    }
+        private ICollection<MsalAccessTokenCacheItem> FilterToAuthoritySpecifiedByAuthenticationRequest(
+            AuthenticationRequestParameters requestParams, ICollection<MsalAccessTokenCacheItem> tokenCacheItems)
+        {
+            if (tokenCacheItems.Count <= 1)
+            {
+                return tokenCacheItems;
+            }
 
-    internal async Task<MsalRefreshTokenCacheItem> FindRefreshTokenAsync(AuthenticationRequestParameters requestParams)
-    {
-        using (ServiceBundle.TelemetryManager.CreateTelemetryHelper(requestParams.RequestContext.TelemetryRequestId, requestParams.RequestContext.ClientId,
-            new CacheEvent(CacheEvent.TokenCacheLookup) { TokenType = CacheEvent.TokenTypes.RT }))
+            requestParams.RequestContext.Logger.Info(
+                "Filtering by authority specified in the authentication request parameters...");
+
+            ICollection<MsalAccessTokenCacheItem> authorityCacheMatches = tokenCacheItems.Where(
+                item => item.Authority.Equals(
+                    requestParams.Authority.CanonicalAuthority,
+                    StringComparison.OrdinalIgnoreCase)).ToList();
+
+            return authorityCacheMatches.Count == 0 ? authorityCacheMatches : tokenCacheItems;
```

Comment:
>return authorityCacheMatches.Count == 0 ? authorityCacheMatches : tokenCacheItems; [](start = 12, length = 82)

i think this statement should be inverted... I wonder if the utests still passes...

Reply:
seems like you changed both the validation and the return param... the U test should be catching this, is a utest missing?

---
In reply to: [239711595](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/708#discussion_r239711595) [](ancestors = 239711595)